workspaces/cli-shared/src/diffs/interaction-diff-worker-rust.ts

```diff
@@ -54,8 +58,13 @@ export class InteractionDiffWorkerRust {
   private async setup() {
     const ignoreFilter = parseIgnore(this.config.ignoreRules);
 
-    const interactionFilter = (i: any) => {
-      return !ignoreFilter.shouldIgnore(i.request.method, i.request.path);
+    const interactionFilter = (interaction: IHttpInteraction): boolean => {
+      const shouldIgnore = ignoreFilter.shouldIgnore(
+        interaction.request.method,
+        interaction.request.path
+      );
+      const isValid = isValidHttpInteraction(interaction);
+      return !shouldIgnore && isValid;
     };
 
     const interactionIterator = CaptureInteractionIterator(
```

Comment:
Ok I realize now that there's a couple of other `CaptureInteractionIterator` - I'm not sure, but should we also add / check if it's a valid http interaction in the capture interaction iterator? Other locations are https://github.com/opticdev/optic/search?q=CaptureInteractionIterator

Reply:
Looking at this more, I think it's probably correct to exclude any invalid http interaction by default here https://github.com/opticdev/optic/blob/develop/workspaces/cli-shared/src/captures/avro/file-system/interaction-iterator.ts#L55

I think by definition, it's something we shouldn't capture - unless we expand our UI / system to capture more explicitly? Thoughts?